### PR TITLE
Lesson 2 - done

### PR DESCRIPTION
**Pull request by balakov-yevhen**
Please, check

### CODE REVIEW

File: Lesson_2/Program.cs [hunk 1]
```diff
@@ -1,12 +1,200 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Text;
 
 namespace Lesson_2
 {
     class Program
     {
+        private static void ClearCurrentConsoleLine()
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs:9]: `Console.Clear()` ?

File: Lesson_2/Program.cs [hunk 2]
```diff
@@ -1,12 +1,200 @@
...
+            int currentLineCursor = Console.CursorTop;
+            Console.SetCursorPosition(0, Console.CursorTop);
+            for (int i = 0; i < Console.WindowWidth; i++)
+                Console.Write(" ");
+            Console.SetCursorPosition(0, currentLineCursor);
+        }
+
+        private static double Summ(double A, double B) 
+        {
+            if (A > B) return A + B;
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs]: Formatting.
`if()`
`{}`

File: Lesson_2/Program.cs [hunk 3]
```diff
@@ -1,12 +1,200 @@
...
+            if (A == B) return A * B;
+            if (A < B) return A - B;
+            throw new Exception("Something went wrong!");
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs]: after ifs you should have empty line

Comment (SamoilenkoS) [reply Lesson_2/Program.cs]: Also this code is unreachable :)

File: Lesson_2/Program.cs [hunk 4]
```diff
@@ -1,12 +1,200 @@
...
+        }
+
+        private static string PointCoordinates(double x, double y)
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs]: Use int as response

File: Lesson_2/Program.cs [hunk 5]
```diff
@@ -1,12 +1,200 @@
...
+                    throw new Exception("Something went wrong!");
+            }
+        }
+
+        private static void SortThreeDigits(ref double first, ref double second, ref double third)
+        {
+            double buffer;
+            if (first > second)
+            {
+                buffer = second;
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs]: Create swap func and use it

File: Lesson_2/Program.cs [hunk 6]
```diff
@@ -1,12 +1,200 @@
...
+                second = first;
+                first = buffer;
+            }
+        }
+
+        private static (double x1, double x2) QuadraticEquation(double a, double b, double c)
+        {
+            double x1, x2;
+            double discriminant = b * b - 4 * a * c;
+            if (discriminant < 0) throw new Exception("Equation has NO roots");
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs]: Formatting

File: Lesson_2/Program.cs [hunk 7]
```diff
@@ -1,12 +1,200 @@
...
+            else
+            {
+                if (discriminant == 0) x1 = x2 = -b / (2 * a);
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs]: Formatting

File: Lesson_2/Program.cs [hunk 8]
```diff
@@ -1,12 +1,200 @@
...
+                first = buffer;
+            }
+        }
+
+        private static (double x1, double x2) QuadraticEquation(double a, double b, double c)
+        {
+            double x1, x2;
+            double discriminant = b * b - 4 * a * c;
+            if (discriminant < 0) throw new Exception("Equation has NO roots");
+            else
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs:87]: Useless if
You can unify it

File: Lesson_2/Program.cs [hunk 9]
```diff
@@ -1,12 +1,200 @@
...
+                }
+                return (x1, x2);
+            }
+        }
+
+        private static string WriteNumber(int num)
+        {
+            string result = null;
+
+            Dictionary<int, string> dictinary = new Dictionary<int, string>
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs]: Here is forbid to use advanced data storage structs.

File: Lesson_2/Program.cs [hunk 10]
```diff
@@ -1,12 +1,200 @@
...
+                { 80, "eighty " },
+                { 90, "ninety " }
+            };
+
+            if (num == 10) result = "ten";
+            if (num == 11) result = "eleven";
+            if (num == 12) result = "twelve";
+            if (num == 13) result = "thirteen";
+            if (num == 15) result = "fifteen";
+            if (num - 10 >= 6 && num - 10 <= 9 || num == 14) result = dictinary[num % 10] + "teen";
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs]: To complicated and incorrect

File: Lesson_2/Program.cs [hunk 11]
```diff
@@ -1,12 +1,200 @@
...
+                { 30, "thirty " },
+                { 40, "forty " },
+                { 50, "fifty " },
+                { 60, "sixty " },
+                { 70, "seventy " },
+                { 80, "eighty " },
+                { 90, "ninety " }
+            };
+
+            if (num == 10) result = "ten";
```
Comment (SamoilenkoS) [inline Lesson_2/Program.cs]: You can use switch

**Review by SamoilenkoS — CHANGES_REQUESTED** [review]

File: Lesson_2/Program.cs (hunk 1 above)
Comment (balakov-yevhen) [reply Lesson_2/Program.cs:9]: You may change current menu item by up or down arrow key. That's why i have to clean line in console

File: Lesson_2/Program.cs (hunk 2 above)
Comment (balakov-yevhen) [reply Lesson_2/Program.cs]: In case of one step {} is not required.

File: Lesson_2/Program.cs (hunk 3 above)
Comment (balakov-yevhen) [reply Lesson_2/Program.cs]: > 

Don't understand, why?

Comment (balakov-yevhen) [reply Lesson_2/Program.cs]: unreachable  - ok, it's for case if user made a mistake in incoming data

File: Lesson_2/Program.cs (hunk 5 above)
Comment (balakov-yevhen) [reply Lesson_2/Program.cs]: it's one more func. case of you question about "cleanconsole" func.. I don't sure if i may to use additional funcs?

File: Lesson_2/Program.cs (hunk 7 above)
Comment (balakov-yevhen) [reply Lesson_2/Program.cs]: don't understand, {} etc?

File: Lesson_2/Program.cs (hunk 8 above)
Comment (balakov-yevhen) [reply Lesson_2/Program.cs:87]: one func for Exceptions ?

File: Lesson_2/Program.cs (hunk 10 above)
Comment (balakov-yevhen) [reply Lesson_2/Program.cs]: what line is incorrect?

File: Lesson_2/Program.cs (hunk 4 above)
Comment (balakov-yevhen) [reply Lesson_2/Program.cs]: what an order for names of sectors in ints?

**Pull request merged**